src/components/LanguagePicker/useLanguagePicker.tsx

```diff
@@ -0,0 +1,187 @@
+import { useEffect, useRef, useState } from "react"
+import { useRouter } from "next/router"
+import { useTranslation } from "next-i18next"
+import { useDisclosure, type UseDisclosureReturn } from "@chakra-ui/react"
+
+import type {
+  I18nLocale,
+  Lang,
+  LocaleDisplayInfo,
+  ProjectProgressData,
+} from "@/lib/types"
+
+import { MatomoEventOptions, trackCustomEvent } from "@/lib/utils/matomo"
+import { languages } from "@/lib/utils/translations"
+
+import progressData from "@/data/translationProgress.json"
+
+import { DEFAULT_LOCALE } from "@/lib/constants"
+
+const data = progressData as ProjectProgressData[]
```

Comment:
I'd keep using `progressData` for its name, much clearer than the generic `data`.

We do have `resolveJsonModule` in our tsconfig so, this data should be typed already. If we need to force the `ProjectProgressData` type, we could do that in our `global.d.ts` using `declare module`.